src/query_plan/operation.rs

```diff
@@ -387,6 +388,27 @@ impl NormalizedSelection {
             }
         }
     }
+
+    pub(crate) fn conditions(&self) -> Result<Conditions, FederationError> {
+        let self_conditions = Conditions::from_directives(self.directives())?;
+        if let Conditions::Boolean(false) = self_conditions {
+            // Never included, there is no point recuring
+            Ok(Conditions::Boolean(false))
+        } else {
+            match self {
+                NormalizedSelection::Field(_) => {
+                    // We want this field regardless of whether sub-selections are included
+                    Ok(self_conditions)
```

Comment:
The logic here doesn't match the JS codebase; when the field has a selection set and the field's conditions are non-boolean, then they should be merged into the selection set's conditions and returned (otherwise you can return `self_conditions`).

Reply:
I’ve changed to match the JS code. However the JS comment says:

> If it is a field, […] no matter what the sub-selection is, we need to get that field

Why does this only apply to boolean conditions?

Reply:
That's a good point! Looking at the callsites of this code, it's used to inform whether query plan nodes should be wrapped in condition nodes (i.e., whether fetches should be skipped entirely). Given that, the sub-selections of a field should indeed not affect whether the field needs to be queried.

This appears to be a bug in query planning code. I've pushed a commit to revert back to your fix, and left a `PORT_NOTE` mentioning the bug. Will merge the code in afterwards!